### PR TITLE
PRO-2399 port

### PR DESCRIPTION
**Pull request by abea**
Please ensure your pull request follows these guidelines:

- [x] Link to the related issue with requirements and/or clearly describe 1) what problem this solves and 2) the requirements to review it against.
- [x] Update the `CHANGELOG.md` file with the added feature or fix. If there is not yet a heading for an upcoming release, add one following the established pattern.
- [x] Keep the pull request minimal! Break large updates into separate pull requests for individual features/fixes whenever possible. Small requests get reviewed more quickly.
- [x] All tests must pass, including the linters. Run `npm run lint` to test code linting independently.

Thanks for contributing!

### CODE REVIEW

File: test-lib/util.js [hunk 1]
```diff
@@ -43,12 +43,14 @@ async function create(options) {
...
+  // TODO: Remove __testDefaults references in 4.x major version or formalize
+  // intended usage with documentation.
   if (!config.__testDefaults) {
     config.modules = config.modules || {};
     const express = config.modules['@apostrophecms/express'] || {};
     express.options = express.options || {};
-    // Allow OS to choose open port
-    express.options.port = null;
+    // Allow OS to choose open port if not explicitly set.
+    express.options.port = express.options.port || null;
```
Comment (abea) [inline test-lib/util.js:53]: If `null` is not special here I can simply delete these lines.

Comment (boutell) [reply test-lib/util.js:53]: `null` is special here. `undefined` defaults to 3000.

Comment (boutell) [reply test-lib/util.js:53]: See the `@apostrophecms/express` module:

```
if (process.env.PORT) {
          port = process.env.PORT;
        } else if (fs.existsSync(self.apos.rootDir + '/data/port')) {
          // Stagecoach option
          port = fs.readFileSync(self.apos.rootDir + '/data/port', 'UTF-8').replace(/\s+$/, '');
        } else {
          port = self.options.port;
        }

        if (port === undefined) {
          port = 3000;
        }

        if (typeof port === 'string') {
          port = Number.isNaN(parseInt(port)) ? port : parseInt(port);
        }
```

**Review by boutell — APPROVED** [review]
Correct as written, we need to continue to pass explicit null when we're not specifying a port, otherwise the `@apostrophecms/express` module defaults to 3000.

**Review by boutell — DISMISSED** [review]

**Review by boutell — APPROVED** [review]

**Pull request merged**